src/services/EcoBridge/Connext/Connext.selectors.ts

```diff
@@ -0,0 +1,192 @@
+
+      const receivingTransaction =
+        transaction.receivingChainId === transaction.chainId ? transaction : partnerTransaction
+
+      const token = CONNEXT_TOKENS.find(
+        tx =>
+          tx.contracts[transaction.sendingChainId] &&
+          tx.contracts[transaction.sendingChainId].contract_address.toLowerCase() ===
+            transaction.sendingAssetId.toLowerCase()
+      )
+
+      const summary: BridgeTransactionSummary = {
+        assetName: token?.symbol ?? '',
+        bridgeId: bridgeId,
+        fromChainId: Number(transaction.sendingChainId),
+        toChainId: Number(transaction.receivingChainId),
+        log: [
+          { chainId: Number(sendingTransaction?.chainId), txHash: sendingTransaction?.prepareTransactionHash ?? '' },
+        ],
+        txHash: sendingTransaction?.prepareTransactionHash ?? '',
+        value: formatUnits(
+          sendingTransaction?.amount ?? 0,
+          token?.contracts[sendingTransaction?.chainId ?? '1'].contract_decimals
+        ),
+        assetAddressL1: transaction.sendingAssetId,
+        assetAddressL2: transaction.receivingAssetId,
+        status: BridgeTransactionStatus.PENDING,
+        timestampResolved: Number(sendingTransaction?.preparedTimestamp) * 1000,
+        pendingReason: 'Transaction has not been confirmed yet',
+      }
+
+      if (partnerTransaction) {
+        if (
+          partnerTransaction.status === ConnextTransactionStatus.PREPARED &&
+          transaction.status === ConnextTransactionStatus.PREPARED
+        ) {
+          summary.status = BridgeTransactionStatus.REDEEM
+          summary.timestampResolved = undefined
+        }
+        if (
+          partnerTransaction.status === ConnextTransactionStatus.FULFILLED &&
+          transaction.status === ConnextTransactionStatus.FULFILLED
+        ) {
+          summary.status = BridgeTransactionStatus.CLAIMED
+
+          summary.log.push({
+            chainId: Number(receivingTransaction?.chainId),
+            txHash: receivingTransaction?.fulfillTransactionHash ?? '',
+          })
+
+          summary.timestampResolved = Number(receivingTransaction?.fulfillTimestamp) * 1000
+        }
+        if (
+          partnerTransaction.status === ConnextTransactionStatus.CANCELLED ||
+          transaction.status === ConnextTransactionStatus.CANCELLED
+        ) {
+          summary.status = BridgeTransactionStatus.CANCELLED
+          summary.log.push({
+            chainId: Number(receivingTransaction?.chainId),
+            txHash: receivingTransaction?.cancelTransactionHash ?? '',
+          })
+          summary.timestampResolved = Number(receivingTransaction?.cancelTimestamp) * 1000
+        }
+        if (
+          transaction.status === ConnextTransactionStatus.PENDING ||
+          partnerTransaction.status === ConnextTransactionStatus.PENDING
+        ) {
+          summary.status = BridgeTransactionStatus.PENDING
+        }
```

Comment:
I would switch ifs. `CANCELLED` can be the last one to avoid case when `summary.status` is overwritten by `PENDING`

Reply:
fixed [ff567c5](https://github.com/levelkdev/swapr-dapp/pull/1008/commits/ff567c5812bcc6d4a3a88e02e87707d5e92ea544)